### DISCUSSION TITLE
Load certificates csr and crl

### DISCUSSION THREAD

**Discussion by adriantom** [FreeOpcUa/opcua-asyncio#1194] [Q&A]
Greetings!

I would like to create server with encryption using crl and csr but as I can see server.load_certificate() method loads only x509 certificate. Is that possible somehow to do that or, if not, is there a plan to add such a functionality (cryptography package makes it possible)?

**Comment by adriantom** [comment]
Bump a question.

Scenario is to have root certificate with crl:
    _\trusted\certs\root.cacerts.der
    \trusted\crl\root.cacrl.pem_
and intermediate certs:
    _\issuers\certs\intermediate.cacerts.der
    \issuers\crl\intermediate.cacrl.pem_
then I would like to deploy those certificates to a server.
Is that scenario possible to realize? To have certificates chain like root -> intermediate with crl's on a server?

**Reply by schroeder-** [reply]
I don't get your problem the server certificate is just used for the client to validate the server identity. The server certificate is just send to the clients. So the client has to validate the certificate. So the client has to have the csr and clr infos stored in his certificate store.

**Reply by schroeder-** [reply]
Or do you want to validate client certificates with the crls and csr? This is currently not implemented, but would be possible to add.

**Reply by adriantom** [reply]
Second case - validate certificates. Is there a plan to add this in the near future?

**Reply by schroeder-** [reply]
There is no plan atm. But this is definitely a feature I consider implementing.
